File: js/tests/palindrome.test.js
const checkPalindrome = require("../palindrome");

const data = [
  ["mom", true],
  ["scary", false],
  ["reviver", true],
  ["stressed", false],
  ["good", false],
  ["refer", true],
  ["something", false],
  ["redder", true],
  ["civic", true]
];

describe("checkPalindrome", () => {
  describe("Return boolean if word is a palindrome", () => {
    test.each(data)("checkPalindrome(%p) = %s", (str, result) => {
      var actual = checkPalindrome(str);

      expect(actual).toEqual(result);
    });
  });
});

File: CSharp/Tests/PalindromeTest.cs
using Xunit;

namespace CSharp.Tests
{
    public class PalindromeTest
    {
        [Theory]
        [InlineData("mom", true)]
        [InlineData("scary", false)]
        [InlineData("reviver", true)]
        [InlineData("stressed", false)]
        [InlineData("good", false)]
        [InlineData("refer", true)]
        [InlineData("something", false)]
        [InlineData("redder", true)]
        [InlineData("civic", true)]
        public void CheckPalindrome_StringInputValue_ReturnBooleanIfWordIsPalindrome(string str, bool expected)
        {
            var actual = Palindrome.CheckPalindrome(str);

            Assert.Equal(expected, actual);
        }
    }
}